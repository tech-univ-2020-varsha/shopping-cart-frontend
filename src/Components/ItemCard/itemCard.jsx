import React, { useEffect } from 'react';
import propTypes from 'prop-types';
import axios from 'axios';
import * as styles from './itemCard.module.css';
import URL from '../../constants/url';
import useItemCount from '../../hooks/useItemCount/useItemCount';


const ItemCard = ({
  id, name,
  price, quantity,
  imageLink, category,
  total, setTotal, sold,
}) => {
  const [count, setCount, callComplete] = useItemCount(id);

  const updateCart = async () => {
    const payload = {
      products: [
        {
          id,
          name,
          price,
          quantity: count,
          imageLink,
          category,
        },
      ],
    };
    const response = await axios({
      method: 'PUT',
      url: `${URL}/cart`,
      data: payload,
    });

    console.log(response);
  };

  useEffect(() => {
    updateCart();
  }, [count]);

  const increment = async () => {
    if (count < quantity) {
      await setCount(count + 1);
      await setTotal(total + 1);
    }
  };


  const decrement = async () => {
    if (count > 0) {
      await setCount(count - 1);
      await setTotal(total - 1);
    }
  };
  if (!callComplete) { return <div>Loading! ...</div>; }
  return (
    <div className={styles.cardContainer}>
      <div className={styles.itemImage}>
        <img src={imageLink} alt={name} className={styles.image} />
      </div>
      <div>
        {name}
      </div>
      <div>
        MRP
        {' '}
        {price}
        /-
      </div>
      {
          !sold
            ? (
              <div className={styles.counter}>
                <div className={styles.counterItems}>
                  <button className={styles.button} onClick={decrement} type="submit">-</button>
                </div>
                <div>
                  {' '}
                  {count}
                </div>
                <div className={styles.counterItems}>
                  <button className={styles.button} onClick={increment} type="submit">+</button>
                </div>
              </div>
            )
            : <button type="submit" className={styles.soldOut}>SOLD OUT</button>

      }

    </div>
  );
};


ItemCard.propTypes = {
  id: propTypes.number.isRequired,
  name: propTypes.string.isRequired,
  price: propTypes.number.isRequired,
  quantity: propTypes.number.isRequired,
  imageLink: propTypes.string.isRequired,
  category: propTypes.string.isRequired,
  total: propTypes.number.isRequired,
  setTotal: propTypes.func.isRequired,
  sold: propTypes.bool.isRequired,
};
export default ItemCard;
