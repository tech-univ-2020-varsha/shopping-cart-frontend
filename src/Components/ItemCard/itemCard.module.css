.cardContainer{
    display: flex;
    flex-direction: column;
    box-shadow: 1px 3px 5px #888888;
    text-align: center;
    margin: 2em;
    padding: 2em;
    border-radius: 20px;
    font-size: 1em;
}

.itemImage{
    border: 5px solid black;
    border-radius: 20px;
}

.image{
    width: 100%;
    height: 100%;
    border-radius: 20px;
}
