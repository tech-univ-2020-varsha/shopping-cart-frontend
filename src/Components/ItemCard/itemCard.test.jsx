import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ItemCard from './itemCard';

describe('The Item Card component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ItemCard
      id={1}
      name="name"
      price={1}
      quantity={1}
      imageLink="imageLink"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
