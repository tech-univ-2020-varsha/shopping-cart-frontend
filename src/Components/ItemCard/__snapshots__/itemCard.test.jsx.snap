// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <img
        alt="name"
        src="imageLink"
      />
    </div>
    <div>
      name
    </div>
    <div>
      MRP 1/-
    </div>
    <div>
      <div>
        <button
          type="submit"
        >
          -
        </button>
      </div>
      <div>
         0
      </div>
      <div>
        <button
          type="submit"
        >
          +
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
